### ISSUE TITLE
Using RDF inside an instance, teleports you randomly

### ISSUE THREAD

**Issue by jamadaha** [azerothcore/azerothcore-wotlk#7388]
### What client do you play on?

enGB

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour
Original report: https://github.com/chromiecraft/chromiecraft/issues/1360

If you are inside an instance, lets take WC; Fishing. Then your queue pops for an instance (Would imagine this also goes for BGS (Not Confirmed). After completing said dungeon you will be teleported out to a seemingly random spot in the world.

Hard to explain please see link below.
https://www.youtube.com/watch?v=nNqNzq8LluU

### Expected Blizzlike Behaviour

Not really sure, I would assume it teleported you to the entrance of the previous instance, in this case WC.

### Steps to reproduce the problem

1. Enter an instance.
2. Go into RDF from said instance (Edited from original report)
3. Complete dungeon
4. Magic.

### Extra Notes
Testers notes:

I don't know whether it should teleport to prior instance, but random teleport seems wrong. At least, you should be teleported to the entrance of the prior instance.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/36a1777b99c400484537c358af8fa6b3a42b13c7

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-CustomWorldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-TempAnnouncements](https://github.com/55Honey/Acore_TempAnnouncements)

### Customizations

None

### Server

ChromieCraft

**Comment by UltraNix** [comment]
It's just a nearest graveyard. You cannot teleprot back to that instance, because it's already unlinked and destroyed (blizzlike). The question is - how it worked on retail? Teleport to instance entrance?

**Comment by jamadaha** [comment]
@UltraNix Are you sure it's the nearest graveyard? I entered a RDF (into the dungeon Slave Pens in Zangarmarsh) from the Deadmines, but leaving the instance I was sent to Shadow Grave. As seen below. On the other hand, I did it before on an orc, which sent me to Durotar. It seems more like it's the start position of that race.
![image](https://user-images.githubusercontent.com/53914190/130283710-561707cd-8886-495d-bed2-d09135d809b4.png)
![image](https://user-images.githubusercontent.com/53914190/130283756-e8b2d3e7-1a1b-4b7d-bb36-2b3e7f2756dd.png)
![image](https://user-images.githubusercontent.com/53914190/130283777-10807b78-fcf6-4ec3-a563-4461a94d7840.png)
